lib/IntegrationConfiguration.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const yaml = require('js-yaml');
+const fs = require('fs');
+
+let integrationPath;
+let privateKeyPath;
+
+/**
+     * Constructor for Asset Compute integration configuration
+     *
+     * @param {string} [integrationFile=] Configuration file path for integration (optional)
+     *                                    If no file path is passed in, ENV will be checked
+     * @param {string} [privateKeyFile=] Configuration file path for private key file when supplying JSON (optional)
+     *                                    If no file path is passed in, ENV will be checked
+     */
+async function getConfiguration(integrationFile, privateKeyFile) {
+
+    privateKeyPath = privateKeyFile || process.env.ASSET_COMPUTE_PRIVATE_KEY_FILE_PATH;
+    integrationPath = integrationFile || process.env.ASSET_COMPUTE_INTEGRATION_FILE_PATH;
+
+    return validate();
+}
+
+function validate() {
+
+    if (!fs.existsSync(integrationPath)) {
+        throw new Error('Missing required files');
+    }
+
+    const data = validateJson() || validateYaml();
+    validateData(data);
+    return data;
+}
+
+function validateData(data) {
+    if ((!data.metascopes || data.metascopes.length === 0)
+            || (!data.technicalAccount || data.technicalAccount.length === 0)
+            || !data.technicalAccount.id
+            || !data.technicalAccount.org
+            || !data.technicalAccount.clientId
+            || !data.technicalAccount.clientSecret
+            || !data.technicalAccount.privateKey
+    ) {
+        throw new Error("Not all integration configuration properties are present");
+    }
+
+}
+
+function validateJson() {
+
+    let json;
+
+    try {
+        json = JSON.parse(fs.readFileSync(integrationPath, 'utf8'));
+    } catch (error) { // eslint-disable-line no-unused-vars
+        return;
+    }
+
+    json.privateKey = validatePrivateKeyFile();
+    return convert(json);
```

Comment:
I like conversion function to indicate what they convert `from` and `to`. Can you change the name so it is obvious?

Reply:
I need ideas O_O the long name would be
`convertIOConsoleJsonToAssetComputeYamlConfiguration()`

I think `convertJsonToYaml()` is a bit confusing because it's not referring to what json it's converting from and we aren't really making a yaml file, just returning a differently structured object that resembles asset compute's yaml integration configuration file as an object

Reply:
`convertConfigJsonToYaml` would be good enough for me!